e2e/baseFixtures.js

```diff
@@ -103,25 +103,40 @@ const extendedTest = test.extend({
    * Default: `true`
    */
   failOnConsoleError: [true, { option: true }],
+  ignore404s: [[], { option: true }],
   /**
    * Extends the base page class to enable console log error detection.
    * @see {@link https://github.com/microsoft/playwright/discussions/11690 Github Discussion}
    */
-  page: async ({ page, failOnConsoleError }, use) => {
+  page: async ({ page, failOnConsoleError, ignore404s }, use) => {
     // Capture any console errors during test execution
-    const messages = [];
+    let messages = [];
     page.on('console', (msg) => messages.push(msg));
-
+    
     await use(page);
 
+    if (ignore404s.length > 0){
+      messages = messages.filter(msg => {
+        let keep = true;
+
+        if (msg.text().match(/404 \(Object Not Found\)/) !== null) {
+          keep = ignore404s.every(ignoreRule => {
+            return msg.location().url.match(ignoreRule) === null;
+          });
+        }
+
+        return keep;
+      });
+    }
+
     // Assert against console errors during teardown
     if (failOnConsoleError) {
-      messages.forEach((msg) =>
+      messages.forEach(async (msg) => {
```

Comment:
there's no await within scope here, so the async block wont make any difference

Reply:
Ooops, that's a hangover from some debugging. Good catch, Thank you!